packages/engine-core/src/data-proxy/DataProxyEngine.ts

```diff
@@ -210,12 +210,16 @@ export class DataProxyEngine extends Engine<DataProxyTxInfoPayload> {
           headers['X-transaction-id'] = interactiveTransaction.id
         }
 
-        const response = await request(url, {
-          method: 'POST',
-          headers: { ...headers, ...this.headers },
-          body: JSON.stringify(body),
-          clientVersion: this.clientVersion,
-        })
+        const response = await request(
```

Comment:
Maybe it could have been simpler like `customFetch(request)(url, ...)` instead of passing a param